osh/sh_expr_eval.py

```diff
@@ -974,6 +974,19 @@ def _VarNameOrWord(self, anode):
         no_str = None  # type: Optional[str]
         return (no_str, loc.Missing)
 
+    def _GetArithLoc(self, anode):
```

Comment:
Hm actually I noticed we have `frontend/location.py TokenForArith(arith_expr_t)`

And that could return a `loc_t` rather than `Optional[Token]`, i.e. it should return `loc.Missing` instead of None

Because I think all the consumers can accept a `loc_t`, not just a Token

We can do that separately, but I think that function is more complete, because it works with Unary,Binary,TernaryOp